test/arrayops.jl

```diff
@@ -1196,9 +1195,33 @@ end
     @test o == fill(1, 3, 4)
 
     # issue #18524
-    m = mapslices(x->tuple(x), [1 2; 3 4], dims=1)
-    @test m[1,1] == ([1,3],)
-    @test m[1,2] == ([2,4],)
+    # m = mapslices(x->tuple(x), [1 2; 3 4], dims=1) # fails, see variations below
+    # @test m[1,1] == ([1,3],)
+    # @test m[1,2] == ([2,4],)
+
+    # issue #21123
+    @test mapslices(nnz, sparse(1.0I, 3, 3), dims=1) == [1 1 1]
```

Comment:
Tests like this must move to https://github.com/JuliaSparse/SparseArrays.jl now, right? So I will remove them here.

Reply:
So that I don't lose them, they are:
```
    # issue #21123
    @test mapslices(nnz, sparse(1.0I, 3, 3), dims=1) == [1 1 1]

    r = rand(Int8, 4,5,2)
    @test vec(mapslices(repr, r, dims=(2,1))) == map(repr, eachslice(r, dims=3))
    @test mapslices(prod, sparse(r[:,:,1]), dims=1) == prod(r[:,:,1], dims=1)
    @test mapslices(cumsum, sparse(r[:,:,1]), dims=1) == cumsum(r[:,:,1], dims=1)
```